src/java/fr/paris/lutece/plugins/workflow/modules/formspdf/service/task/FormsPDFTask.java

```diff
@@ -104,8 +112,9 @@ public void processTask( int nIdResourceHistory, HttpServletRequest request, Loc
             {
                 user = AdminUserService.getAdminUser( request );
             }
+            
             HtmlToPDFGenerator htmltopdf = new HtmlToPDFGenerator( form.getTitle( ), I18nService.getLocalizedString( PROPERTY_LABEL_DESCRIPTION, locale ), frep,
-                    formsPDFTaskConfig.getTemplate( ) );
+            		FormsPDFTaskTemplateHome.findByPrimaryKey(formsPDFTaskConfig.getIdTemplate()), provider );
```

Comment:
The FormsProvider is only needed to get a collection of markers, maybe the HTMLToPDFGenerator should only ask for this collection, for example :

collectionNotifyMarkers= GenericFormsProvider.getProviderMarkerDescriptions( form );

the FormsProvider class would then be useless.
What do you think ?